sdk/src/wallet/wallet/operations/storage.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(feature = "storage")]
+mod storage_stub {
+    use alloc::sync::Arc;
+
+    use async_trait::async_trait;
+    use serde::{Deserialize, Serialize};
+    use tokio::sync::RwLock;
+
+    use crate::{
+        client::{
+            secret::{mnemonic::MnemonicSecretManager, SecretManage, SecretManagerConfig},
+            storage::StorageAdapter,
+            ClientBuilder,
+        },
+        wallet::{
+            storage::constants::{SECRET_MANAGER_KEY, WALLET_INDEXATION_KEY},
+            wallet::builder::StorageOptions,
+            WalletBuilder,
+        },
+    };
+
+    #[derive(Debug, Serialize, Deserialize)]
+    struct WalletData {
+        client_options: Option<ClientBuilder>,
+        coin_type: Option<u32>,
+        storage_options: Option<StorageOptions>,
+    }
+
+    impl WalletData {
+        fn into_builder<S: SecretManage>(self, secret_manager: Option<S>) -> WalletBuilder<S> {
+            WalletBuilder {
+                client_options: self.client_options,
+                coin_type: self.coin_type,
+                storage_options: self.storage_options,
+                secret_manager: secret_manager.map(|s| Arc::new(RwLock::new(s))),
+            }
+        }
+    }
+
+    #[async_trait]
+    pub trait SaveLoadWallet {
+        async fn save_data(
+            &self,
+            storage: &impl StorageAdapter<Error = crate::wallet::Error>,
+        ) -> crate::wallet::Result<()>;
+
+        async fn get_data(
```

Comment:
Considering the trait is named **SaveLoad**Wallet, maybe `load_data` is the more expected name for this fn?

Reply:
+1